framework/meta-lib/src/tools/wasm_extractor.rs

```diff
@@ -49,25 +49,35 @@ fn populate_wasm_info(
     let mut allocator_trigger = false;
     let mut ei_check = false;
     let mut memory_grow_flag = false;
-    let mut has_panic = "none";
+    let mut has_panic: PanicMessage = PanicMessage::default();
 
     let parser = Parser::new(0);
     for payload in parser.parse_all(&wasm_data) {
         match payload? {
             Payload::ImportSection(import_section) => {
                 imports = extract_imports(import_section, extract_imports_enabled);
-                ei_check = is_ei_valid(imports.clone(), check_ei);
+                ei_check |= is_ei_valid(imports.clone(), check_ei);
             },
             Payload::DataSection(data_section) => {
-                allocator_trigger = is_fail_allocator_triggered(data_section.clone());
-                if is_panic_with_message_triggered(data_section.clone()) {
-                    has_panic = WITH_MESSAGE;
-                } else if is_panic_without_message_triggered(data_section) {
-                    has_panic = WITHOUT_MESSAGE;
+                allocator_trigger |= is_fail_allocator_triggered(data_section.clone());
+                match has_panic {
```

Comment:
This match belongs to a method in `PanicMessage`. Something like `max_severity`. We should also have a separate function that gives the PanicMessage for the current section.

We should also rename it to something like PanicReport or PanicStatus, since it is a flag, rather than just a message.